cmd/config/internal/commands/run-fns.go

```diff
@@ -68,31 +81,63 @@ func (r *RunFnRunner) runE(c *cobra.Command, args []string) error {
 	return handleError(c, r.RunFns.Execute())
 }
 
-// getFunctions parses the commandline flags and arguments into explicit
+// getContainerFunctions parses the commandline flags and arguments into explicit
 // Functions to run.
-func (r *RunFnRunner) getFunctions(c *cobra.Command, args, dataItems []string) (
+func (r *RunFnRunner) getContainerFunctions(c *cobra.Command, args, dataItems []string) (
 	[]*yaml.RNode, error) {
-	// if image isn't specified, then Functions is empty
-	if r.Image == "" {
+	if r.Image == "" && r.StarPath == "" {
 		return nil, nil
 	}
 
-	// create the function spec to set as an annotation
-	fn, err := yaml.Parse(`container: {}`)
-	if err != nil {
-		return nil, err
-	}
-	// TODO: add support network, volumes, etc based on flag values
-	err = fn.PipeE(
-		yaml.Lookup("container"),
-		yaml.SetField("image", yaml.NewScalarRNode(r.Image)))
-	if err != nil {
-		return nil, err
+	var fn *yaml.RNode
+	var err error
+
+	// if image isn't specified, then Functions is empty
+	if r.Image != "" {
+		// create the function spec to set as an annotation
+		fn, err = yaml.Parse(`container: {}`)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: add support network, volumes, etc based on flag values
+		err = fn.PipeE(
+			yaml.Lookup("container"),
+			yaml.SetField("image", yaml.NewScalarRNode(r.Image)))
+		if err != nil {
+			return nil, err
+		}
+		if r.Network {
+			err = fn.PipeE(
+				yaml.LookupCreate(yaml.MappingNode, "container", "network"),
+				yaml.SetField("required", yaml.NewScalarRNode("true")))
+			if err != nil {
+				return nil, err
+			}
+		}
 	}
-	if r.Network {
+
+	if r.EnableStar && r.StarPath != "" {
+		// if image isn't specified, then Functions is empty
+		if r.StarPath == "" {
```

Comment:
Does this ever happen as the upper if checks r.StarPath != ""

Reply:
Nope